### PR TITLE
Generalisation of hasVN to template variables.

### PR DESCRIPTION
**Pull request by CalebSerafin**
## PR Type
* Bug Fix
* Change
* [x] Enhancement

### Changes
1. Default First Aid Kits and Medkits can be overridden in each faction template.
2. PAVN forces are now correctly given PAVN medical gear.
3. Civilian Uniforms are loaded from the civilian template.
4. configSort and equipmentIsValidForCurrentModset have had their booleans been inverted.
5. If no GreenFor TFAR radios are found: It will fall back to using BluFor and OpFor radios.
6. All TFAR faction encryption codes are the same.
7. All "FakeAmmo" is removed.
8. Vanilla Medkits, ToolKits, ItemMap, FirstAidKit are removed if removed from a faction's equipment via the template override.
9. Dive Gear and Flight Gear can be overridden by templates from the default of true .
10. createAIControls & wavedCA will not attempt to spawn a UAV if the faction has "not_supported" as its UAV.
11. `customUnitTypes` has been prefixed to `A3A_customUnitTypes`
12. getVehicleIntel will not give faction UAV intel if the faction's UAV is a "not_supported".
13. placeIntel loads object types from overridable template defaults. A boolean is provided for intel of whether it is a computer and should have lights & bombs attached.
14. A3A_fnc_itemset_medicalSupplies requires a side identifier when used outside of template creation. This only affects equipRebel.
15. reDress loads uniforms directly from the template without a shim.
16. Faction data variables have been refactored to `"A3A_faction_" + _factionPrefix` where _factionPrefix is either ["occ","inv","reb","civ"].
17. loadFaction supports loading multiple files in order, laterfiles override previous definitions.
18. Undercover uniforms are pulled directly from templates without a shim.
19. A3A_hasVN has been moved lower in the selector to allow other modsets to play on the Cam Lao Nam Terrain.
20. Loaded addon pack names are stored in A3A_LoadedContentAddons.

### Changes failed to be implimented
21. Automatic detection if a faction has UAV based on empty array/classnames. (Support and garrison still need to handle when UAVs are not available)
22. Rename MACV and PAVN faction to have AI Prefix.
23. Allowed Rebel starting arsenal categories from templates.
24. Blacklist Starting equipment mods from templates.
25. Bonus Ace lootItem from templates.

# Further testing required
1. [ ] No
* Yes

### Completed checks by author
* [x] Have you loaded the mission in LAN host?
* [ ] Have you loaded the mission on a dedicated server?

## Testing each change
* [x] 1. Recruit and kill an AI rifleman and AI medic. Rifleman should have 3 FAKs, medic: 10 FAKs and a medikit. If using VN, they should have the VN equivalents. 
* [x] 2. Enemy VN factions should have their own type of medical supplies. VN civs should have PAVN medical supplies.
* [x] 3. Test two different modests, civilians should be given appropriate clothing.
* [x] 4. Most starting equipment in modsets should remain unchanged (except additional PAVN medikit).
* [x] 5. VN should have west and east radios. Other modsets should have independent radios.
* [x] 6. All faction SRs and LRs should be able to communicate 
* [x] 7. There should be no food or vanity items on any modset.
* [x] 8. Vanilla Medkits, ToolKits, ItemMap, FirstAidKit are removed if removed from a faction's equipment via the template override.
* [x] 9. Most modsets should have dive and lfight gear. VN should not.
* [x] 10. createAIControls & wavedCA will not attempt to spawn a UAV if the faction has "not_supported" as its UAV. (Cannot Unit Test)
* [x] 11. No undefined RPT errors about customUnitTypes should be thrown.
* [x] 12. getVehicleIntel will not give faction UAV intel if the faction's UAV is a "not_supported". (Cannot Unit Test)
* [x] 13. All other modsets should have camping table with documents or illuminated bomb laptop. VN should have a wooden table with documents or no bomb photos.
* [x] 14. No RPT errors regardigng A3A_fnc_itemset_medicalSupplies should be thrown. All units are given medical gear.
* [x] 15. All rebel units should have uniforms.
* [x] 16. No RPT errors regarding faction data should be thrown.
* [x] 17. No RPT errors regarding faction data should be thrown.
* [x] 18. Undercover works with only non-military uniforms.
* [x] 19. You can play RHS on Cam Lao Nam. (IDK something else might break along the way)
* [x] 20. If you have either IvoryCars, TCGM_BikeBackPack, D3SCars, RDSCars: use the debug console to see a friendly name in A3A_LoadedContentAddons.

### CODE REVIEW

**Comment by Bob-Murphy** [comment]
Was on the testserver for a few hours, not a single error in rpt

**Review by HakonRydland — COMMENTED** [review]
most if not all commits have been checked by one or more of the dev team while it was being written.

it is therefore unlikely to be broken anywhere at this stage.

File: A3-Antistasi/Templates/NewTemplates/VN/VN_CIV.sqf [hunk 1]
```diff
@@ -1,12 +1,28 @@
...
+    "vn_o_uniform_vc_02_07",
+    "vn_o_uniform_vc_03_02",
+    "vn_o_uniform_vc_04_02",
+    "vn_o_uniform_vc_05_01",
+    "vn_o_uniform_vc_02_05",
+    "vn_o_uniform_vc_04_03",
+    "vn_o_uniform_vc_03_03"
+]] call _fnc_saveToTemplate;
+
+["headgear", [""]] call _fnc_saveToTemplate;
```
Comment (HakonRydland) [inline A3-Antistasi/Templates/NewTemplates/VN/VN_CIV.sqf:17]: turns out ``_unit addHeadgear nil`` dosnt cause issues, and is ignored, therefore we can remove the empty strings from all the headgear arrays in the templates

**Review by HakonRydland — APPROVED** [review]
ok very quick glance, everything i recognised from previous review of singular commits i glanced over, everything else a proper review, found no issues.

on thing that could be changed but wont impact anything if left as is is the headgear defines in the templates, see comment in review

**Review by Bob-Murphy — APPROVED** [review]
Tested

**Pull request merged**